### ISSUE TITLE
feat(eval): support automatic speech recognition metrics

### ISSUE THREAD

**Issue by ssss141414** [microsoft/winml-cli#1229]
## Feature gap

`winml eval` does not register `automatic-speech-recognition`, so valid Wav2Vec2 CTC artifacts cannot be evaluated with a task metric through the CLI.

## Reproduction

- Model: `MahmoudAshraf/mms-300m-1130-forced-aligner`
- Task: `automatic-speech-recognition`
- Artifacts: verified CPU FP32 and FP16 ONNX models with `input_values [1,16000] -> logits [1,49,31]`

Run either:

`winml eval --schema --task automatic-speech-recognition`

or:

`winml eval -m <model.onnx> --model-id MahmoudAshraf/mms-300m-1130-forced-aligner --task automatic-speech-recognition --device cpu --ep cpu`

Both are rejected because ASR is absent from the supported-task registry.

## Requested support

Add an ASR evaluator/schema with raw-audio preprocessing and a defensible transcript/audio dataset contract. For CTC checkpoints, expose an appropriate metric such as WER; forced-alignment checkpoints may additionally need alignment/timestamp quality evaluation.

The referenced model card only says “forced alignment dataset” and does not identify a dataset repository, subset, split, or revision, so callers must be able to provide an explicit evaluation dataset rather than silently substituting one.

**Comment by DingmaomaoBJTU** [comment]
## Triage Summary

**Type:** Feature Request (Enhancement)
**Priority:** P2
**Component:** model / task scale

**Assessment:** This is a valid feature request to add ASR evaluator support to `winml eval`. The implementation would require a new evaluator schema for `automatic-speech-recognition` task, raw-audio preprocessing, and WER metric support. CTC and forced-alignment checkpoints may need distinct evaluation paths.

**Next steps:** Needs owner assignment. Implementation should align with existing evaluator patterns (see #1175 for audio-classification as a reference).

---
*Triaged on 2026-07-31*
